common/environment.go

```diff
@@ -164,6 +165,20 @@ func (EnvironmentVariable) ManagedIdentityResourceString() EnvironmentVariable {
 	}
 }
 
+func (EnvironmentVariable) WorkloadIdentityClientID() EnvironmentVariable {
+	return EnvironmentVariable{
+		Name:        "AZCOPY_WORKLOAD_CLIENT_ID",
+		Description: "Client ID for Workload identity. This variable is only used for auto login, please use the command line flag instead when invoking the login command.",
+	}
+}
+
+func (EnvironmentVariable) WorkloadIdentityTokenFilePath() EnvironmentVariable {
+	return EnvironmentVariable{
+		Name:        "AZCOPY_WORKLOAD_TOKEN_FILE_PATH",
+		Description: "Token file path for Workload identity. This variable is only used for auto login, please use the command line flag instead when invoking the login command.",
+	}
+}
+
```

Comment:
Client IDs and paths of service account token files for Azure Workload Identity are supposed not to be manually set by users.

Instead, those values are supposed to be automatically set by the runtime environment (the Azure Workload Identity Webhook in AKS environment, for example) as the following environment variables.

| Environment variable | Description |
| -- | -- |
| AZURE_AUTHORITY_HOST | The Azure Active Directory (AAD) endpoint. |
| AZURE_CLIENT_ID | The client ID of the AAD application or user-assigned managed identity. |
| AZURE_TENANT_ID | The tenant ID of the registered AAD application or user-assigned managed identity. |
| AZURE_FEDERATED_TOKEN_FILE | The path of the projected service account token file. |

https://azure.github.io/azure-workload-identity/docs/quick-start.html#7-deploy-workload -> See "Output"

Therefore, we should retrieve a client ID and the path of the service account token file from the env variables `AZURE_CLIENT_ID` and `AZURE_FEDERATED_TOKEN_FILE`, not from `AZCOPY_WORKLOAD_CLIENT_ID` and `AZCOPY_WORKLOAD_TOKEN_FILE_PATH`.

In addition, we should also retrieve a tenant ID from `AZURE_TENANT_ID`.

This is the default behavior implemented by Azure Identity SDK. Example:

https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/workload_identity.go#L42-L53

Reply:
All these environment variables will work by default as we use the Identity SDK under the hood.

Reply:
@adreed-msft This is what I was mentioning in the description of the PR. I think it might be cleaner just to always use environment variables and let the identity SDK handle the reading and parsing of them. Then we can get rid of our environment variables for this and our CLI params. The only thing the user would have to do is either do azcopy login login-type=workload or set the auto login env var to workload